### ISSUE TITLE
Calling token information and renewal

### ISSUE THREAD

**Issue by pvandervelde** [rajanadar/VaultSharp#41]
Is there a way to determine when the calling token (as created by a `VaultClient` instance) needs to be renewed? As far as I can tell there is a `IVaultClient.RenewCallingTokenAsync` method which allows you to renew the token but I can't seem to figure out how to get information about the token so that I can determine what the lease time is etc. Without having the lease time it will be hard to determine when the lease should be renewed.

**Comment by rajanadar** [comment]
In vault, the onus of tracking the expiry time of a token and renewing it is left to the ultimate consumer. Neither the value api nor this library can help you.

As a balance between timers and renew-on-401, you can agree on a policy to have a token be valid for 12 hours or so, and then have your app necessarily renew it blindly after within 10 hours or so.

**Comment by pvandervelde** [comment]
I understand that but if you create a new `VaultClient` via the `VaultClientFactory.CreateVaultClient` then there is no way to get a hold of the token that was created, thus you cannot renew it sensibly. 

Setting a 12 hour period is not really inline with the idea of Vault which is that you are told how long your token is valid (usually a short period of time) so that you can renew it on time.

Would it be sensible to provide an access method for the user to get the token because that token is stored in the `VaultClient` class (https://github.com/rajanadar/VaultSharp/blob/88ead72ed2389373aabf464a8b2c6cf7b5f8505d/src/VaultSharp/VaultClient.cs#L42)

**Comment by rajanadar** [comment]
@pvandervelde 
Just published a new version of VaultSharp 0.10.4xxx

https://www.nuget.org/packages/VaultSharp/0.10.4000

In this redesigned library, I expose a property on every IAuthMethodInfo to see the login-token that was generated.

For 0.6.x, you may need to use reflection to peek into the _lazyVaultToken variable.

**Comment by pvandervelde** [comment]
Awesome! I will upgrade to the latest version of the library and get all the shiny new features. Thanks heaps!

**Comment by rajanadar** [comment]
Thanks. 

I realized that I was waiting too long after a vault api release to release the library with 100% feature set. Instead I have changed the model a bit. The 0.10.4000 release has all the most common features. And as per the need, quick increments of 4001,4002 etc can be published.

Also, I targeted. NET standard 2.0 for this, which may not be the case for the consumers. This library has been completely redesigned and hence trying out new things. There is an item in progress to try for. Net standard 1.x as well.

Let me know how it fits your needs.
